clients/lsp-kotlin.el

```diff
@@ -36,7 +36,12 @@
   'lsp-clients-kotlin-server-executable
   "lsp-mode 6.4")
 
-(defcustom lsp-clients-kotlin-server-executable "kotlin-language-server"
+;; TODO is it okay to reuse this?
```

Comment:
Does changing this variable default value would cause any breaking changes? if a user has a custom value for that variable, with this pr would we break that? if so, it's not a good idea to re-use it, otherwise sounds ok to me

Reply:
If I’ve understood correctly, it shouldn’t, because the custom value will be passed to `lsp-dependency` as the `:binary-path` and it shouldn’t be downloaded again. But what about if I do this, to be safe?

1. Put the generated value I have here in a new variable.
2. Let `lsp-clients-kotlin-server-executable` remain unchanged except to add `.bat` to the default value on Windows. Anyone already setting the variable should be unaffected.
3. Change this form to use `(:system lsp-client-kotlin-server-executable)` if found, and the `:download :url` second.

This way, those who’ve set `lsp-clients-kotlin-server-executable` to a valid executable will see it work automatically, while everyone else (including those who’ve set it to an invalid path) will be prompted to download the language server.

Reply:
Sounds good to me